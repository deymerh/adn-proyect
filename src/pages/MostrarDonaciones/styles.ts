import { createStyles, makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(() =>
  createStyles({
    container: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      height: '100vh',
      flexDirection: 'column',
    },
    table: {
      minWidth: 650,
    },
    text: {
      marginTop: '10px',
      textAlign: 'center',
    },
  }),
);
