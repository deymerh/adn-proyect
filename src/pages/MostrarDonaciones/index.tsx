import React, { useContext } from 'react';
import {
  Typography,
  Grid,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  Button,
} from '@material-ui/core';
import { Link } from 'react-router-dom';
import { useStyles } from './styles';
import { DataContext } from '../../context/DataContext';

const MostrartDonaciones: React.FC = () => {
  const classes = useStyles();
  const { globalState } = useContext(DataContext);
  return (
    <Grid container className={classes.container}>
      <Typography variant="h5" color="textSecondary">
        Información de donantes
      </Typography>
      <Grid justifyContent="center" container item xs={10} md={6}>
        <Grid item md={10} xs={12}>
          <TableContainer component={Paper}>
            <Table className={classes.table} aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>Nombre</TableCell>
                  <TableCell align="right">Edad</TableCell>
                  <TableCell align="right">Fecha</TableCell>
                  <TableCell align="right">Cantidad Donada</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {globalState.personasDonantes.map((item) => (
                  <TableRow key={item.nombre}>
                    <TableCell component="th" scope="row">
                      {item.nombre}
                    </TableCell>
                    <TableCell align="right">{item.edad}</TableCell>
                    <TableCell align="right">{item.fecha}</TableCell>
                    <TableCell align="right">{item.cantidadDonada}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
          </TableContainer>
          <Typography
            variant="h5"
            color="textSecondary"
            className={classes.text}
          >
            Información de fondos generales
          </Typography>
          <TableContainer component={Paper}>
            <Table className={classes.table} aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell align="center">Caridad</TableCell>
                  <TableCell align="center">Contrucción</TableCell>
                  <TableCell align="center">Niños</TableCell>
                  <TableCell align="center">Fondos Total</TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <TableRow>
                  <TableCell align="center" component="th" scope="row">
                    {globalState.caridad}
                  </TableCell>
                  <TableCell align="center">
                    {globalState.construccion}
                  </TableCell>
                  <TableCell align="center">{globalState.ninos}</TableCell>
                  <TableCell align="center">
                    {globalState.dineroTotal}
                  </TableCell>
                </TableRow>
              </TableBody>
            </Table>
          </TableContainer>
        </Grid>
      </Grid>
      <Button size="small">
        <Link to="/">Nueva Donación</Link>
      </Button>
    </Grid>
  );
};
export default MostrartDonaciones;
