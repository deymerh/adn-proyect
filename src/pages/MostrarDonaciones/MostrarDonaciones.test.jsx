import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import MostrarDonaciones from './index';

describe('Pruebas en el <MostrartDonaciones/>', () => {
  const page = render(
    <BrowserRouter>
      <MostrarDonaciones/>
    </BrowserRouter>
  );
  test('Crear el SnapShot del compoente', () => {
    expect(page).toMatchSnapshot();
  });
  test('Que tenga el titulo de los donantes', () => {
  render(
    <BrowserRouter>
      <MostrarDonaciones/>
    </BrowserRouter>
  );
  const text1 = screen.getByText('Información de donantes');
  expect(text1).toBeInTheDocument();
  });

  test('Que tenga el titulo de la cantidad por departamentos', () => {
  render(
    <BrowserRouter>
      <MostrarDonaciones/>
    </BrowserRouter>
  );
  const text2 = screen.getByText('Información de fondos generales');
  expect(text2).toBeInTheDocument();
  });
  
});
