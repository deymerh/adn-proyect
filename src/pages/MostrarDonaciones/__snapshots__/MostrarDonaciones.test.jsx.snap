// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <MostrartDonaciones/> Crear el SnapShot del compoente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-container-1 MuiGrid-container"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
        >
          Información de donantes
        </h5>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
          >
            <div
              class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <table
                aria-label="simple table"
                class="MuiTable-root makeStyles-table-2"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head"
                      scope="col"
                    >
                      Nombre
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                      scope="col"
                    >
                      Edad
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                      scope="col"
                    >
                      Fecha
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                      scope="col"
                    >
                      Cantidad Donada
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                />
              </table>
            </div>
            <h5
              class="MuiTypography-root makeStyles-text-3 MuiTypography-h5 MuiTypography-colorTextSecondary"
            >
              Información de fondos generales
            </h5>
            <div
              class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <table
                aria-label="simple table"
                class="MuiTable-root makeStyles-table-2"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Caridad
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Contrucción
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Niños
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                      scope="col"
                    >
                      Fondos Total
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                      role="cell"
                      scope="row"
                    >
                      0
                    </th>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      0
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      0
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    >
                      0
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              href="/"
            >
              Nueva Donación
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-container-1 MuiGrid-container"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
      >
        Información de donantes
      </h5>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-center MuiGrid-grid-xs-10 MuiGrid-grid-md-6"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
        >
          <div
            class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root makeStyles-table-2"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Nombre
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Edad
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Fecha
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Cantidad Donada
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              />
            </table>
          </div>
          <h5
            class="MuiTypography-root makeStyles-text-3 MuiTypography-h5 MuiTypography-colorTextSecondary"
          >
            Información de fondos generales
          </h5>
          <div
            class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root makeStyles-table-2"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Caridad
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Contrucción
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Niños
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    scope="col"
                  >
                    Fondos Total
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                    role="cell"
                    scope="row"
                  >
                    0
                  </th>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    0
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    0
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
                  >
                    0
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <a
            href="/"
          >
            Nueva Donación
          </a>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
